Language/Haskell/GhcMod/Monad/Types.hs

```diff
@@ -129,10 +138,11 @@ data GhcModState = GhcModState {
       gmGhcSession   :: !(Maybe GmGhcSession)
     , gmComponents   :: !(Map ChComponentName (GmComponent GMCResolved (Set ModulePath)))
     , gmCompilerMode :: !CompilerMode
+    , gmMMappedFiles :: Map TargetId (StringBuffer, UTCTime)
```

Comment:
I'm not sure how that would work. As I see it, there are three options:

1. Map keys can only be filenames
2. Map keys can only be module names
3. Map keys can be either.

For (1) and (2) this is a moot point, and TargetId is just a convenience choice here, since only one constructor is ever used. For (3), it doesn't really matter if it's TargetId or String or otherwise, since duplication is possible in any case, and I don't see how I would extract module name given source code and file name, assuming there can be errors in said code.